overlord/snapstate/handlers.go

```diff
@@ -1622,7 +1622,9 @@ func (m *SnapManager) undoLinkSnap(t *state.Task, _ *tomb.Tomb) error {
 		return err
 	}
 
-	if newInfo.GetType() == snap.TypeSnapd {
+	// only restart here if snapd was the installed and needs to
```

Comment:
We could improve the comment a little to provide a little guidance when we debug the problem again:

```
// restart only when snapd was installed for the first time and the rest of
// the cleanup is performed by snapd from core;
// when reverting a subsequent snapd revision, the restart happens in
// undoLinkCurrentSnap() instead
```

Reply:
I can make this change in my existing PR